scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,530 @@
+func (o EvidenceHandler) AppraiseEvidence(
+	ec *proto.EvidenceContext,
+	endorsementsStrings []string,
+) (*ear.AttestationResult, error) {
+	var (
+		err         error
+		evidenceMap map[string]interface{}
+	)
+
+	refVal, err := refvalToComidTriple(endorsementsStrings)
+	if err != nil {
+		return nil, err
+	}
+
+	evidence, err := evidenceToComidTriple(ec)
+	if err != nil {
+		return nil, err
+	}
+
+	result := handler.CreateAttestationResult(SchemeName)
+
+	appraisal := result.Submods[SchemeName]
+
+	appraisal.TrustVector.InstanceIdentity = ear.NoClaim
+	appraisal.TrustVector.Executables = ear.NoClaim
+	appraisal.TrustVector.Configuration = ear.NoClaim
+	appraisal.TrustVector.FileSystem = ear.NoClaim
+	appraisal.TrustVector.StorageOpaque = ear.NoClaim
+	appraisal.TrustVector.SourcedData = ear.NoClaim
+	appraisal.TrustVector.Hardware = ear.UnsafeHardwareClaim
+	appraisal.TrustVector.RuntimeOpaque = ear.VisibleMemoryRuntimeClaim
+
+claimsLoop:
+	for _, m := range refVal.Measurements.Values {
+		var (
+			k  uint64
+			em *comid.Measurement
+		)
+
+		k, err = m.Key.GetKeyUint()
+		if err != nil {
+			break
+		}
+
+		// We can skip validating certain claims for the following reasons:
+		// - POLICY ToDo: Do we need to test individual policy features?
+		// - CURRENT_TCB is informational only. It's best handled by policy
+		// - PLATFORM_INFO ToDO: Do we need to test individual platform features?
+		// - REPORT_DATA is a nonce supplied by user for freshness, not applicable for verification
+		// - REPORT_ID is ephemeral, so we can't use it for verification.
+		// - REPORT_ID_MA is also ephemeral, used for migration
+		// - CHIP_ID is unique to an specific attester, but reference values could be used more generally
+		// - Current Version (CURRENT_MAJOR/MINOR/BUILD) should already be part of REPORTED_TCB.
+		//     ToDo: It is a good idea to test it anyway, but the Version type only tests for
+		//     equality, and this would trigger spurious failures
+		// - COMMITTED_TCB is informational, used by the host to advance REPORTED_TCB
+		if k == mKeyPolicy ||
+			k == mKeyCurrentTcb ||
+			k == mKeyPlatformInfo ||
+			k == mKeyReportData ||
+			k == mKeyReportID ||
+			k == mKeyReportIDMA ||
+			k == mKeyChipID ||
+			k == mKeyCommittedTcb ||
+			k == mKeyCurrentVersion ||
+			k == mKeyCommittedVersion {
+			continue
+		}
+
+		em, err = measurementByUintKey(*evidence, k)
+		if err != nil {
+			break
+		}
+
+		if em == nil {
+			err = fmt.Errorf("MKey %d not found in Evidence", k)
+			break
+		}
+
+		switch k {
+		case mKeyReportedTcb:
+			if !compareTcb(m, *em) {
+				err = fmt.Errorf("reported TCB in evidence doesn't match reference")
+				break claimsLoop
+			}
+		case mKeyLaunchTcb:
+			reportedTcb, err := measurementByUintKey(*evidence, mKeyReportedTcb)
+			if err != nil {
+				break claimsLoop
+			}
+			if !compareTcb(*reportedTcb, *em) {
+				// ToDo: Is this a failure condition?
```

Comment:
Note: Each of these failures should be represented in the corresponding flag inside the TrustVector for Platform ? If not please check and raise a bug on AR4SI if something fundamental missing in the spec..